utils/bincompat/generate.py

```diff
@@ -0,0 +1,289 @@
+            stream.write("-device virtio-net-pci,netdev=en0 \\\n")
+            stream.write('    -append "netdev.ipv4_addr=172.44.0.2 ')
+            stream.write("netdev.ipv4_gw_addr=172.44.0.1 ")
+            stream.write("netdev.ipv4_subnet_mask=255.255.255.0 ")
+            stream.write('vfs.fstab=[ \\"initrd:/:initrd:::\\" ] ')
+            stream.write('-- $cmd" \\\n')
+        else:
+            stream.write("    -append ")
+            stream.write('vfs.fstab=[ \\"initrd:/:initrd:::\\" ] ')
+            stream.write('-- $cmd" \\\n')
+        stream.write('    -initrd "$PWD"/rootfs.cpio \\\n')
+        stream.write("    -cpu max\n")
+        stbuf = os.stat(out_file)
+        os.chmod(out_file, stbuf.st_mode | stat.S_IEXEC)
+
+
+def generate_run_kraft(config, plat):
+    """Generate running script using KraftKit."""
+
+    out_file = os.path.join(RUNDIR, f"kraft-run-{plat}.sh")
+    with open(out_file, "w", encoding="utf8") as stream:
+        stream.write(RUN_KRAFT_HEADER)
+        stream.write(RUN_KILL_COMMANDS)
+        if config["networking"]:
+            stream.write(RUN_COMMON_NET_COMMANDS)
+            stream.write(RUN_KRAFT_NET_COMMANDS)
+        stream.write("\n")
+        if config["networking"]:
+            stream.write("sudo ")
+        stream.write(
+            "KRAFTKIT_BUILDKIT_HOST=docker-container://buildkitd kraft run \\\n"
+        )
+        if "accel" not in config.keys():
+            stream.write("    -W \\\n")
+        elif not config["accel"]:
+            stream.write("    -W \\\n")
+        if config["arch"]:
+            stream.write("    -W \\\n")
+        stream.write(f"    --memory {config['memory']}M \\\n")
+        stream.write("    --log-level debug --log-type basic \\\n")
+        if config["networking"]:
+            stream.write("    --network bridge:virbr0 \\\n")
+        stream.write(f"    --arch {config['arch']} --plat {plat}\n")
+
+        stbuf = os.stat(out_file)
+        os.chmod(out_file, stbuf.st_mode | stat.S_IEXEC)
+
+
+def generate_run(config):
+    """Generate running scripts."""
+
+    generate_run_fc_json(config, "base_fc-x86_64")
+    generate_run_fc_json(config, "base_fc-x86_64-strace")
+    generate_run_fc_json(config, "base_fc-x86_64-debug")
+    generate_run_fc(config)
+    generate_run_qemu(config, "base_qemu-x86_64")
+    generate_run_kraft(config, "qemu")
+
+
+def main():
+    """The main program function calls generate functions."""
+
+    # Obtain configurations for running applications.
+    with open(CONFIG, "r", encoding="utf8") as stream:
+        config = yaml.safe_load(stream)
+        if not "rootfs" in config.keys():
+            config["rootfs"] = "rootfs"
+
+    # Currently only x86_64 is supported.
+    config["arch"] = "x86_64"
+
+    # Obtain targets for KraftKit runs form kraft.yaml.
+    with open(KRAFTCONFIG, "r", encoding="utf8") as stream:
+        data = yaml.safe_load(stream)
+
+        config["name"] = data["name"]
```

Comment:
Is `name:` required in the `Kraftfile`?
Should this be

```py
if not "name" in data.keys():
    config["name"] = directory name
```

Reply:
@StefanJum , I added a check for that. Yes, `name` will be required in `Kraftfile`.